src/main/java/net/kodehawa/mantarobot/commands/ModerationCmds.java

```diff
@@ -306,55 +306,59 @@ protected void call(Context ctx, String content, String[] args) {
-                    return;
+                String memberRaw = args[0];
 
-                User user = member.getUser();
+                ctx.findMember(memberRaw, ctx.getMessage()).onSuccess(members -> {
+                    Member member = CustomFinderUtil.findMember(memberRaw, members, ctx);
+                    if (member == null)
+                        return;
 
-                if (!ctx.getMember().canInteract(member)) {
-                    ctx.sendLocalized("commands.kick.hierarchy_conflict", EmoteReference.ERROR);
-                    return;
-                }
+                    User user = member.getUser();
 
-                if (ctx.getAuthor().getId().equals(user.getId())) {
-                    ctx.sendLocalized("commands.kick.yourself_note", EmoteReference.ERROR);
-                    return;
-                }
+                    if (!ctx.getMember().canInteract(member)) {
+                        ctx.sendLocalized("commands.kick.hierarchy_conflict", EmoteReference.ERROR);
+                        return;
+                    }
 
-                //If one of them is in a higher hierarchy than the bot, cannot kick.
-                if (!selfMember.canInteract(member)) {
-                    ctx.sendLocalized("commands.kick.self_hierarchy_conflict", EmoteReference.ERROR2, user.getName());
-                    return;
-                }
-                final DBGuild db = MantaroData.db().getGuild(ctx.getGuild());
+                    if (ctx.getAuthor().getId().equals(user.getId())) {
+                        ctx.sendLocalized("commands.kick.yourself_note", EmoteReference.ERROR);
+                        return;
+                    }
 
-                guild.kick(member).reason(finalReason).queue(
-                        success -> {
-                            if (!user.isBot()) {
-                                user.openPrivateChannel()
-                                        .flatMap(privateChannel ->
-                                                privateChannel.sendMessage(String.format("%sYou were **kicked** by %s#%s with reason: %s on server **%s**.",
-                                                EmoteReference.MEGA, ctx.getAuthor().getName(), ctx.getAuthor().getDiscriminator(), finalReason, ctx.getGuild().getName()))
-                                        ).queue();
-                            }
+                    //If one of them is in a higher hierarchy than the bot, cannot kick.
+                    if (!selfMember.canInteract(member)) {
+                        ctx.sendLocalized("commands.kick.self_hierarchy_conflict", EmoteReference.ERROR2, user.getName());
+                        return;
+                    }
+                    final DBGuild db = MantaroData.db().getGuild(ctx.getGuild());
 
-                            db.getData().setCases(db.getData().getCases() + 1);
-                            db.saveAsync();
+                    guild.kick(member).reason(finalReason).queue(
```

Comment:
DMing should be done before kicking, since this might be the last mutual guild

Reply:
Yup, moved it up. We do permission checks before anyway, so it's very unlikely to fail.